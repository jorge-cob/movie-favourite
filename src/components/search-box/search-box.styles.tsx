import styled from 'styled-components';

export const BoxContainer = styled.div`
  height: 40px;
  width: 100%;
  display: flex;
  justify-content: center;
  margin-bottom: 25px;
`;

export const SearchButtonItem = styled.button`
  margin-left: 10px;
`;
