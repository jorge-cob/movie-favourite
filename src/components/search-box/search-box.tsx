import React from 'react';

interface SearchProps {
  onSearchChange: React.ChangeEventHandler<HTMLInputElement>;
  onSearchClick(): any;
}

export const SearchBox = ({onSearchChange, onSearchClick}: SearchProps) => (
  <>
    <input
      className='search-box'
      type='search'
      placeholder='search movies'
      onChange={onSearchChange}
    />
    <button className='search-button' onClick={onSearchClick}>
      Search
    </button>
  </>
);
