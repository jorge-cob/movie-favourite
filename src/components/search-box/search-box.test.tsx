import React from 'react';
import { shallow } from 'enzyme';

import { SearchBox } from './search-box';

describe('SearchBox component', () => {

  const onSearchChange = jest.fn();
  const onSearchClick = jest.fn();

  const wrapper = shallow(<SearchBox onSearchChange={onSearchChange} onSearchClick={onSearchClick} />);

  it('should render SearchBox component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onSearchChange when text changes', () => {
    wrapper.find('.search-box').simulate('change');
    expect(onSearchChange).toHaveBeenCalled();
  });
  it('should call onSearchClick when button is clicked', () => {
    wrapper.find('.search-button').simulate('click');
    expect(onSearchClick).toHaveBeenCalled();
  });
});