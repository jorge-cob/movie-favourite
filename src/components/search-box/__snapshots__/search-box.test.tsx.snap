// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropdown component should render CartDropdown component 1`] = `ShallowWrapper {}`;

exports[`CartDropdown component should render SearchBox component 1`] = `ShallowWrapper {}`;
